BUILD.bazel

```diff
@@ -191,7 +191,54 @@ config_setting(
 )
 
 container_bundle(
-    name = "build-other-images",
+    name = "build-other-images_s390x",
+    images = {
+        # cmd images
+        "$(container_prefix)/$(image_prefix)sidecar-shim:$(container_tag)": "//cmd/sidecars:sidecar-shim-image",
+        # container-disk images
+        "$(container_prefix)/$(image_prefix)alpine-container-disk-demo:$(container_tag)": "//containerimages:alpine-container-disk-image",
+        # Kernel boot container
+        # -
+        # Customized container-disk images
+        "$(container_prefix)/$(image_prefix)fedora-with-test-tooling-container-disk:$(container_tag)": "//containerimages:fedora-with-test-tooling",
+        # testing images
+        "$(container_prefix)/$(image_prefix)vm-killer:$(container_tag)": "//images/vm-killer:vm-killer-image",
+    },
+)
+
+container_bundle(
+    name = "build-other-images_aarch64",
+    images = {
```

Comment:
Can we define a variable such as
```
standard_images = {....}
```

and then use it for the archs...
```
container_bundle(
    name = "build-other-images_aarch64",
    images =  standard_images,
)
....
```

Reply:
I'm afraid i'm not an expert with bazel and what i find with google regarding variables just points to build variables from env. If there is an option to make a placeholder/variable that would of course make this change much more simple.

Reply:
```
diff --git a/BUILD.bazel b/BUILD.bazel
index 9d2ce692bb..f417e91d3a 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -190,10 +190,8 @@ config_setting(
     values = {"define": "release=true"},
 )
 
-container_bundle(
-    name = "build-other-images",
-    images = {
-        # cmd images
+
+standard_images = {        # cmd images
         "$(container_prefix)/$(image_prefix)sidecar-shim:$(container_tag)": "//cmd/sidecars:sidecar-shim-image",
         "$(container_prefix)/$(image_prefix)example-hook-sidecar:$(container_tag)": "//cmd/sidecars/smbios:example-hook-sidecar-image",
         "$(container_prefix)/$(image_prefix)example-disk-mutation-hook-sidecar:$(container_tag)": "//cmd/sidecars/disk-mutation:example-disk-mutation-hook-sidecar-image",
@@ -216,7 +214,12 @@ container_bundle(
         "$(container_prefix)/$(image_prefix)nfs-server:$(container_tag)": "//images/nfs-server:nfs-server-image",
         "$(container_prefix)/$(image_prefix)vm-killer:$(container_tag)": "//images/vm-killer:vm-killer-image",
         "$(container_prefix)/$(image_prefix)winrmcli:$(container_tag)": "//images/winrmcli:winrmcli-image",
-    },
+}
+
+
+container_bundle(
+    name = "build-other-images",
+    images = standard_images,
 )
```
Basically I am asking for this, the `standard_images` can then be used across multiple targets (x86, amd,...).  You can go as far as having a dict for "cmd images" and such and than merge them. It just makes this easier to maintain and spot the difference

Reply:
Ok, thank you for the explanation, i will try it out.

Reply:
I have pushed https://github.com/kubevirt/kubevirt/commit/6cbea391b96ad2cd11a9744c82e75f0a938ade64 to a new branch for testing.
What i found so far, i can use the variable as is for s390x, since it contains all the images. However i'm not sure how to re-use it for x86 and aarch64, since i need to add additional images. All i'm getting is syntax errors, as it seems to be expecting a new image definition and not a variable:
```
ERROR: /root/go/src/kubevirt.io/kubevirt/BUILD.bazel:214:35: syntax error at ',': expected :
ERROR: /root/go/src/kubevirt.io/kubevirt/BUILD.bazel:242:35: syntax error at ',': expected :
ERROR: /root/go/src/kubevirt.io/kubevirt/vendor/mvdan.cc/sh/v3/cmd/shfmt/BUILD.bazel:21:10: While resolving toolchains for target //vendor/mvdan.cc/sh/v3/cmd/shfmt:shfmt: invalid registered toolchain '//:py_toolchain': no such target '//:py_toolchain': target 'py_toolchain' not declared in package '' defined by /root/go/src/kubevirt.io/kubevirt/BUILD.bazel
ERROR: Analysis of target '//vendor/mvdan.cc/sh/v3/cmd/shfmt:shfmt' failed; build aborted:
INFO: Elapsed time: 0.196s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (2 packages loaded, 0 targets configured)
FAILED: Build did NOT complete successfully (2 packages loaded, 0 targets configured)
    currently loading: @bazel_tools//tools/jdk ... (12 packages)
    Fetching @local_jdk; fetching
make: *** [Makefile:212: format] Error 1
```

Reply:
@jschintag This should help
```
diff --git a/BUILD.bazel b/BUILD.bazel
index 9d2ce692bb..a98aa93a5e 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -190,9 +190,8 @@ config_setting(
     values = {"define": "release=true"},
 )
 
-container_bundle(
-    name = "build-other-images",
-    images = {
+
+defualt_images = {
         # cmd images
         "$(container_prefix)/$(image_prefix)sidecar-shim:$(container_tag)": "//cmd/sidecars:sidecar-shim-image",
         "$(container_prefix)/$(image_prefix)example-hook-sidecar:$(container_tag)": "//cmd/sidecars/smbios:example-hook-sidecar-image",
@@ -215,8 +214,14 @@ container_bundle(
         "$(container_prefix)/$(image_prefix)disks-images-provider:$(container_tag)": "//images/disks-images-provider:disks-images-provider-image",
         "$(container_prefix)/$(image_prefix)nfs-server:$(container_tag)": "//images/nfs-server:nfs-server-image",
         "$(container_prefix)/$(image_prefix)vm-killer:$(container_tag)": "//images/vm-killer:vm-killer-image",
-        "$(container_prefix)/$(image_prefix)winrmcli:$(container_tag)": "//images/winrmcli:winrmcli-image",
-    },
+}
+
+
+load("@bazel_skylib//lib:dicts.bzl", "dicts")
+
+container_bundle(
+    name = "build-other-images",
+    images = dicts.add(defualt_images,{"$(container_prefix)/$(image_prefix)winrmcli:$(container_tag)": "//images/winrmcli:winrmcli-image"}),
 )

```

Reply:
Note that I have `defualt_images` variable defined and then I can just use it in function as `dicts.add(default_images, ...)`

Reply:
Thank you @xpivarc that was the missing link. I have pushed my changes with the variable to this PR.